projects/js-packages/charts/src/components/line-chart/line-chart.tsx

```diff
@@ -17,11 +17,42 @@ import { withResponsive } from '../shared/with-responsive';
 import styles from './line-chart.module.scss';
 import type { BaseChartProps, DataPointDate, SeriesData } from '../../types';
 
+type CurveType = 'smooth' | 'linear' | 'monotone';
+
 const X_TICK_WIDTH = 100;
 
+/**
+ * Determines the curve type for the line chart based on the provided type and smoothing parameters
+ *
+ * @param {CurveType} type      - The explicit curve type to use
+ * @param {boolean}   smoothing - Legacy smoothing parameter
+ * @return The curve function to use for the line
+ */
+const getCurveType = ( type?: CurveType, smoothing?: boolean ) => {
```

Comment:
I would prefer the old switch... case.. format, which is more efficient especially for multiple conditions. I was thinking we just move the testing of `smooth` to the beginning of the function like this old commit:

https://github.com/Automattic/jetpack/pull/42281/commits/ef396648fe25b9bffe0f8002d579f459c35ff115

Reply:
Let's focus on the videopress changes first and then come back to this one 😄

Reply:
OK I have refreshed this with the old switch-case format, with smoothing check first - similar to that old commit linked. Let me know what you think of this approach :)